### PR TITLE
Fix testRecoveryIsCancelledAfterDeletingTheIndex

### LINKED ISSUE #76560: [CI] SnapshotBasedIndexRecoveryIT testRecoveryIsCancelledAfterDeletingTheIndex failing
This is a weird timeout issue. From what I can tell there are no related issues. Seems suspect that its on arm.

**Build scan:**
https://gradle-enterprise.elastic.co/s/exqmn7xel4bcg/tests/:server:internalClusterTest/org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT/testRecoveryIsCancelledAfterDeletingTheIndex

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT.testRecoveryIsCancelledAfterDeletingTheIndex" -Dtests.seed=E36E1308685B0AF6 -Dtests.locale=th-TH-u-nu-thai-x-lvariant-TH -Dtests.timezone=Indian/Reunion -Druntime.java=16`

**Applicable branches:**
7.x

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT&tests.test=testRecoveryIsCancelledAfterDeletingTheIndex

**Failure excerpt:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.

  at __randomizedtesting.SeedInfo.seed([E36E1308685B0AF6]:0)

```

### PR DESCRIPTION
**Pull request by henningandersen**
Closes #76560

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java [hunk 1]
```diff
@@ -496,6 +490,12 @@ public void testRecoveryIsCancelledAfterDeletingTheIndex() throws Exception {
                 }
             );
 
+            assertAcked(
```
Comment (arteam) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java:493]: Is the difference between these two snippets is that we have to register a send behaviour before updating indices settings?

Comment (henningandersen) [reply server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java:493]: yes, if we update settings first, we risk the recovery starting or completing before we add the send behavior. That makes the `recoverSnapshotFileRequestReceived.await()` a few lines down never complete, failing the test.

Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java:493]: We had an inadvertent race between the call to `addRequestHandlingBehavior` and the end of the recovery triggered by the update settings call - if the recovery completed before we got around to adding the request handler then we'd never count down the `recoverSnapshotFileRequestReceived` latch. The fix is to hook into the transport service before we even start the recovery.

**Pull request merged**